packages/smooth_app/lib/pages/product/add_new_product/add_new_product_type.dart

```diff
@@ -0,0 +1,226 @@
+                        ),
+                        shape: BoxShape.circle,
+                      ),
+                      child: Opacity(
+                        opacity: _opacityAnimation.value,
+                        child: const icons.Check(
+                          size: 9.0,
+                        ),
+                      ),
+                    ),
+                    const SizedBox(width: MEDIUM_SPACE),
+                    Expanded(
+                      child: Padding(
+                        padding: const EdgeInsetsDirectional.only(
+                          bottom: LARGE_SPACE,
+                        ),
+                        child: Column(
+                          crossAxisAlignment: CrossAxisAlignment.start,
+                          children: <Widget>[
+                            Text(
+                              widget.productType.getTitle(appLocalizations),
+                              overflow: TextOverflow.ellipsis,
+                              maxLines: 1,
+                              style: TextStyle(
+                                fontWeight: FontWeight.bold,
+                                color: lightTheme ? Colors.black : Colors.white,
+                              ),
+                            ),
+                            const SizedBox(height: SMALL_SPACE),
+                            Text(
+                              widget.productType.getSubtitle(appLocalizations),
+                              overflow: TextOverflow.ellipsis,
+                              style: TextStyle(
+                                color: lightTheme
+                                    ? Colors.black54
+                                    : Colors.white54,
+                              ),
+                            ),
+                          ],
+                        ),
+                      ),
+                    )
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  void _initAnimationsIfNecessary(bool lightTheme) {
+    if (_controller != null) {
+      return;
+    }
+
+    _currentTheme = lightTheme;
+
+    _controller = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 300),
+    )..addListener(() => setState(() {}));
+
+    final ThemeData themeData = Theme.of(context);
+
+    _colorAnimation = ColorTween(
+      begin: themeData.scaffoldBackgroundColor.withOpacity(0.0),
+      end: lightTheme
+          ? themeData.extension<SmoothColorsThemeExtension>()!.primaryMedium
+          : themeData.extension<SmoothColorsThemeExtension>()!.primarySemiDark,
+    ).animate(
+      CurvedAnimation(parent: _controller!, curve: Curves.fastOutSlowIn),
+    );
+
+    _opacityAnimation = Tween<double>(begin: 0.0, end: 1.0).animate(
+      CurvedAnimation(parent: _controller!, curve: Curves.fastOutSlowIn),
+    );
+  }
+}
+
+extension ProductTypeExtension on ProductType {
```

Comment:
Given that I prefer coherence, please create a specific file called product_type_extension.dart and merge your code with the specific one in product_query.dart.